packages/protocol/contracts/layer1/shasta/impl/Inbox.sol

```diff
@@ -136,10 +136,13 @@ abstract contract Inbox is EssentialContract, IInbox {
         coreState = _finalize(config, coreState, claimRecords);
 
         // Verify capacity for new proposals
-        _verifyCapacity(config, coreState);
+        uint256 availableCapacity = _getAvailableCapacity(config, coreState);
+        require(availableCapacity > 0, ExceedsUnfinalizedProposalCapacity());
 
-        // Process forced inclusion if required
-        coreState = _processForcedInclusion(config, coreState);
+        if (availableCapacity > 1) {
```

Comment:
I agree the capacity check should account for the forced inclusion as well, so this is a good fix. But now it creates the possibility that the forced inclusion is postponed indefinitely if the buffer is full and proposals are created at the rate they are proven. I'd recommend insisting that the forced inclusion is the next proposal once it's due, so if there is a forced inclusion, the chain would only progress where there are two slots available.